ring.go

```diff
@@ -653,6 +653,39 @@ func (c *Ring) Close() error {
 	return c.shards.Close()
 }
 
+func (c *Ring) Watch(fn func(*Tx) error, keys ...string) error {
+	if len(keys) == 0 {
+		return fmt.Errorf("redis: Watch requires at least one key")
+	}
+
+	var shards []*ringShard
+	for _, key := range keys {
+		if key != "" {
+			shard, err := c.shards.GetByKey(hashtag.Key(key))
+			if err != nil {
+				return err
+			}
+
+			shards = append(shards, shard)
+		}
+	}
+
+	if len(shards) == 0 {
+		return fmt.Errorf("redis: Watch requires at least one shard")
+	}
+
+	if len(shards) > 1 {
+		for _, shard := range shards[1:] {
+			if shard.Client.String() != shards[0].Client.String() {
```

Comment:
Can we use `shard.Client != shards[0].Client` or `shard.Client.Options() != shards[0].Client.Options()`? Relying on `String` does not look right...

Reply:
I will use the first, anyway `String` is unique on host/port, that should be what we care only for, that's why I used it in the first place. maybe I'm missing something